include/swift/AST/DiagnosticsParse.def

```diff
@@ -131,17 +131,17 @@ ERROR(lex_unicode_escape_braces,none,
 ERROR(lex_illegal_multiline_string_start,none,
       "multi-line string literal content must begin on a new line", ())
 ERROR(lex_illegal_multiline_string_end,none,
-      "multi-line string literal content must not be on the delimiter's line", ())
-ERROR(lex_insufficient_string_indent,none,
-      "string literal content is insufficiently indented", ())
-ERROR(lex_unexpected_tab_in_string_indent,none,
-      "string literal content is indented with a tab where a space is expected", ())
-ERROR(lex_unexpected_space_in_string_indent,none,
-      "string literal content is indented with a space where a tab is expected", ())
+      "multi-line string literal closing delimiter must begin on a new line", ())
+ERROR(lex_inconsistent_string_indent,none,
+      "line does not start with same indentation as closing delimiter of multi-line string literal", ())
+NOTE(note_insufficient_string_indent,none,
+      "unexpectedly found non-indentation character here", ())
+NOTE(note_unexpected_char_in_string_indent,none,
+      "unexpected %select{space|tab}0 in indentation here", (bool))
+NOTE(note_matching_indentation_char_here,none,
+      "should match %select{space|tab}0 in same position before closing delimiter", (bool))
```

Comment:
Excellent analysis, and great point about suppressing messages past a certain point. I would only quibble with the suggestion that these messages should all be errors.

The more I think about it, the more it seems to me that each (uninterpolated) multiline string literal should at most give rise to one error for indentation. After all, in every case the problem (well, the syntactic problem) is curable by deleting all whitespace preceding the closing delimiter. Of course, that may not give you the _intended_ string, and we wouldn't necessarily want to have that as a fix-it for people to mash on, but I think it does demonstrate that this is conceptually one error (with perhaps multiple places to be fixed, appropriate for notes) and not multiple.